src/libdrakvuf/drakvuf.c

```diff
@@ -349,3 +349,65 @@ status_t drakvuf_get_struct_member_rva(const char *rekall_profile,
                 rva,
                 NULL);
 }
+
+bool drakvuf_get_kern_module_base_addr( drakvuf_t drakvuf, char *module_name, addr_t *base_addr )
+{
+    size_t name_len = strlen( module_name );
+    addr_t kernel_list_head;
+    vmi_instance_t vmi = drakvuf->vmi;
+
+    vmi_pause_vm(vmi);
+
+    vmi_read_addr_ksym( vmi, "PsLoadedModuleList", &kernel_list_head );
+
+    addr_t next_module = kernel_list_head;
+
+    while( 1 )
+    {
+        addr_t tmp_next = 0;
+
+        if ( vmi_read_addr_va( vmi, next_module, 4, &tmp_next ) != VMI_SUCCESS )
+            break;
+
+        if (kernel_list_head == tmp_next)
+            break;
+
+        *base_addr = 0 ;
+
+        vmi_read_addr_va( vmi, next_module + offsets[LDR_DATA_TABLE_ENTRY_DLLBASE], 4, base_addr );
+
+        if ( ! *base_addr )
+            break;
+
+        unicode_string_t *us = vmi_read_unicode_str_va( vmi, next_module + offsets[LDR_DATA_TABLE_ENTRY_BASEDLLNAME], 4 );
+
+        if ( us )
+        {
+            unicode_string_t out = { 0 };
+            if ( vmi_convert_str_encoding( us, &out, "UTF-8" ) == VMI_SUCCESS  )
+            {
+                if ( ! strncasecmp( (char *)out.contents, module_name, name_len ) )
+                {
+                    free( out.contents );
+                    vmi_free_unicode_str( us );
+                    return true ;
+                }
+
+                if ( out.contents ) 
+                    free( out.contents );
+            }
+            vmi_free_unicode_str( us );
+        }
+
+        next_module = tmp_next ;
+    }
+
+    return false ;
+}
+
+
+void drakvuf_add_trap_dll( drakvuf_t drakvuf, drakvuf_trap_t *trap, addr_t vaddr, vmi_pid_t pid )
```

Comment:
...and I don't have a computer right in front of me to check it but It seems to me that "inject_trap()" was unavailable to be used from the plugins, was it?